### PR TITLE
[promise-cpp] Add new port

### PR DESCRIPTION
**Pull request by chausner**
**Describe the pull request**

- #### What does your PR fix?  
  Adds new port [promise-cpp](https://github.com/xhawk18/promise-cpp)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run ./vcpkg x-add-version --all and committed the result?  
  Yes

### CODE REVIEW

File: ports/promise-cpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+file(INSTALL ${SOURCE_PATH}/LICENSE.md DESTINATION ${CURRENT_PACKAGES_DIR}/share/promise-cpp RENAME copyright)
```
Comment (JonLiu1993) [inline ports/promise-cpp/portfile.cmake:17]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/promise-cpp" RENAME copyright)
```

File: ports/promise-cpp/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "promise-cpp",
+  "version-string": "1.0.3",
```
Comment (JonLiu1993) [inline ports/promise-cpp/vcpkg.json:3]:
```suggestion
  "version": "1.0.3",
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
This is error log:
[install-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/7230644/install-x64-osx-dbg-out.log)

**Comment by chausner** [comment]
I updated the port for the new major version 2.1.0 which was released yesterday. Unfortunately, the CMake install targets have apparently be removed from CMakeLists.txt so I now simply copy the headers into `include`.

The suggestion by vcpkg is not ideal, though:
>The package promise-cpp is header only and can be used from CMake via:
>
>    find_path(PROMISE_CPP_INCLUDE_DIRS "promise/add_ons.hpp")
>    target_include_directories(main PRIVATE ${PROMISE_CPP_INCLUDE_DIRS})

`add_ons.hpp` is not the primary header file. Is there any way to change what header file it recommends?

**Comment by dg0yt** [comment]
> Is there any way to change what header file it recommends?

Add a `usage` file with the rectified usage instructions.

**Comment by chausner** [comment]
> > Is there any way to change what header file it recommends?
> 
> Add a `usage` file with the rectified usage instructions.

Thanks, added a usage file.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by chausner** [comment]
> Would you consider it acceptable to install into a subidrectory like <promise-cpp/promise.hpp>?

I noticed there is also another issue that is preventing the port from being usable right now: https://github.com/xhawk18/promise-cpp/blob/5e533a1c2beb363ca6b87f7f002007388349366c/include/promise.hpp#L262

I will open an issue and ask the maintainer if they can address these issues in the library itself so we don't have to work around them with patches.

**Comment by BillyONeal** [comment]
> I noticed there is also another issue that is preventing the port from being usable right now: https://github.com/xhawk18/promise-cpp/blob/5e533a1c2beb363ca6b87f7f002007388349366c/include/promise.hpp#L262

Ouch! Best of luck!

**Review by PhoebeHui — APPROVED** [review]
LGTM, thanks for your contribution!

**Comment by BillyONeal** [comment]
Thanks for running that down with upstream!

**Pull request merged**